src/AspNetCoreModuleV2/RequestHandlerLib/PipeOutputManager.cpp

```diff
@@ -2,19 +2,22 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 #include "stdafx.h"
+#include "exceptions.h"
+#include <mutex>
 
 PipeOutputManager::PipeOutputManager() :
     m_dwStdErrReadTotal(0),
     m_hErrReadPipe(INVALID_HANDLE_VALUE),
     m_hErrWritePipe(INVALID_HANDLE_VALUE),
-    m_hErrThread(INVALID_HANDLE_VALUE),
-    m_fDisposed(FALSE)
+    m_hErrThread(INVALID_HANDLE_VALUE)
 {
 }
 
 PipeOutputManager::~PipeOutputManager()
 {
-    StopOutputRedirection();
+    // For some reason, we can't put StopOutputRedirection as the argument here.
```

Comment:
This doesn't work. If you call StopOutputRedirection destructor would call it again.

Reply:
I see. Two things I can do:
- Make StopOutputRedirection private
- StopOutputRedirection just does a double check lock to set a bool.

I'll do the second one for now.